src/typescript/frontend/src/app/dexscreener/pair.ts

```diff
@@ -0,0 +1,115 @@
+ * `uniswap`, but if multiple DEXes are tracked an id such as a factory address may be used.
+ * - `asset0` and `asset1` order should **never** change. `amount0/reserve0` will always refer to the same `asset0`,
+ * and `amount1/reserve1` will always refer to the same `asset1`. If asset order mutates then all data after the change
+ * will be invalid and a re-index will be required.
+ *     - A simple strategy to keep this in check is to simply order assets alphabetically. For example, in a pair
+ * containing assets `0xAAA` and `0xZZZ`, `asset0=0xAAA` and `asset1=0xZZZ`
+ *     - DEX Screener UI will automatically invert pairs as needed and default to their most logical order (i.e.:
+ * `BTC/USD` as opposed to `USD/BTC`)
+ * - `createdAtBlockNumber`, `createdAtBlockTimestamp` and `createdAtTxnId` are optional but encouraged. If unavailable
+ * DEX Screener can bet set to assume pair creation date is the same date as its first ever event.
+ * - `feeBps` corresponds to swap fees in bps. For instance, a fee of 1% maps to `feeBps=100`
+ * - `pool` is only recommended for DEXes that support multi-asset pools and allows the DEX Screener UI to correlate
+ * multiple pairs in the same multi-asset pool
+ * - `metadata` includes any optional auxiliary info not covered in the default schema and not required in most cases
+ */
+export interface Pair {
+  id: string;
+  dexKey: string;
+  asset0Id: string;
+  asset1Id: string;
+  createdAtBlockNumber?: number;
+  createdAtBlockTimestamp?: number;
+  createdAtTxnId?: string;
+  creator?: string;
+  feeBps?: number;
+  pool?: {
+    id: string;
+    name: string;
+    assetIds: string[];
+    pairIds: string[];
+    metadata?: Record<string, string>;
+  };
+  metadata?: Record<string, string>;
+}
+
+export interface PairResponse {
+  pair: Pair;
+}
+
+/**
+ *
+ * @param pairId is the pair ID. Generally it's `event.market.symbolEmojis.join("") + "-APT"`
+ */
+export async function getPair(pairId: string): Promise<Pair> {
+  const emojiString = pairId.split("-")[0];
+  const sybolEmojis = emojiString.split("");
+
+  const symbolBytesArray = sybolEmojis.map((e) => SYMBOL_EMOJI_DATA.byEmoji(e)!.bytes);
+  const symbolBytes = new Uint8Array(symbolBytesArray.flatMap((v) => Array.from(v)));
+
+  const marketRegistrations = await fetchMarketRegistrationEventBySymbolBytes({ symbolBytes });
+  const marketRegistration = marketRegistrations[0];
+
+
+  const aptos = getAptosClient();
+  const block = await aptos.getBlockByVersion({ ledgerVersion: marketRegistration.transaction.version });
+
+  return {
+    id: pairId,
+    dexKey: "emojicoin.fun",
+    asset0Id: emojiString,
+    asset1Id: "APT",
+    createdAtBlockNumber: parseInt(block.block_height),
+    createdAtBlockTimestamp: marketRegistration.transaction.timestamp.getTime() / 1000,
+    createdAtTxnId: String(marketRegistration.transaction.version),
+    // TODO: validate there is no swap fee (is there?)
+    feeBps: 0,
```

Comment:
There's an optional swap fee based on the `integratorFeeRateBPs` passed to the contract. It's optional if you call `swap` directly and pass `0`, otherwise we use a fee rate of `INTEGRATOR_FEE_RATE_BPS` for swaps. I think in this case, if we're linking directly to the emojicoin.fun app, the feeBps should be indicated as `INTEGRATOR_FEE_RATE_BPS`, since that's the fee they will incur when interacting with the swap part of the contract